src/zenml/zen_server/routers/stack_deployment_endpoints.py

```diff
@@ -0,0 +1,135 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Endpoint definitions for stack deployments."""
+
+import datetime
+from typing import Optional, Tuple
+
+from fastapi import APIRouter, Request, Security
+
+from zenml.constants import API, INFO, STACK, STACK_DEPLOYMENT, URL, VERSION_1
+from zenml.enums import StackDeploymentProvider
+from zenml.models import DeployedStack, StackDeploymentInfo
+from zenml.stack_deployments.utils import get_stack_deployment_class
+from zenml.zen_server.auth import AuthContext, authorize
+from zenml.zen_server.exceptions import error_response
+from zenml.zen_server.utils import (
+    handle_exceptions,
+)
+
+router = APIRouter(
+    prefix=API + VERSION_1 + STACK_DEPLOYMENT,
+    tags=["stacks"],
+    responses={401: error_response, 403: error_response},
+)
+
+
+@router.get(
+    INFO,
+    response_model=StackDeploymentInfo,
+)
+@handle_exceptions
+def get_stack_deployment_info(
+    provider: StackDeploymentProvider,
+    _: AuthContext = Security(authorize),
+) -> StackDeploymentInfo:
+    """Get information about a stack deployment provider.
+
+    Args:
+        provider: The stack deployment provider.
+
+    Returns:
+        Information about the stack deployment provider.
+    """
+    stack_deployment_class = get_stack_deployment_class(provider)
+    return StackDeploymentInfo(
+        provider=provider,
+        description=stack_deployment_class.description(),
+        instructions=stack_deployment_class.instructions(),
+        post_deploy_instructions=stack_deployment_class.post_deploy_instructions(),
+        permissions=stack_deployment_class.permissions(),
+        locations=stack_deployment_class.locations(),
+    )
+
+
+@router.get(
+    URL,
+)
+@handle_exceptions
+def get_stack_deployment_url(
+    request: Request,
+    provider: StackDeploymentProvider,
+    stack_name: str,
+    location: Optional[str] = None,
+    auth_context: AuthContext = Security(authorize),
+) -> Tuple[str, str]:
```

Comment:
Should we maybe already check here that the user has permissions to create service connectors, stacks and components? Otherwise I assume the lambda function downstream will fail, which leaves users with deployed infra that is not tracked in ZenML?

Reply:
Good point, but not something I can check in the client. I would need a new endpoint just for this purpose and I would need to check all permissions, not just the service connectors. Perhaps I can use the endpoint that @bcdurak already set up. Maybe add a "verify" flag to it.

Reply:
Can;t we have those checks on this exact URL endpoint? Or what is the purpose for creating this URL other than to open it and create stacks?

Reply:
Done